awsremote.py

```diff
@@ -50,57 +55,73 @@ def check_shell(self, command):
+def ip_address_of(instance, ec2):
+    address = ec2.describe_instances(InstanceIds=[instance])
+    # Returns one of the values in the previous call.
+    return address['Reservations'][0]['Instances'][0]['PublicIpAddress']
+
 def get_machines(num_instances_to_use, aws_group_name):
     machines = []
     #connect to AWS
-    ec2 = boto.ec2.connect_to_region('us-west-2');
-    autoscale = boto.ec2.autoscale.AutoScaleConnection();
+    ec2 = boto3.client('ec2');
+    autoscale = boto3.client('autoscaling');
 
     #how many machines are currently running?
-    group = autoscale.get_all_groups(names=[aws_group_name])[0]
-    num_instances = len(group.instances)
-    print(GetTime(),'Number of instances online:',len(group.instances))
+    instances = autoscale.describe_auto_scaling_instances()['AutoScalingInstances']
+    num_instances = len(instances)
+    print(GetTime(),'Number of instances online:', num_instances)
 
     #switch on more machines if we need them
     if num_instances < num_instances_to_use:
         print(GetTime(),'Launching instances...')
-        autoscale.set_desired_capacity(aws_group_name,num_instances_to_use)
+        autoscale.set_desired_capacity(
+            AutoScalingGroupName = aws_group_name,
+            DesiredCapacity = num_instances_to_use
+        )
 
         #tell us status every few seconds
-        group = None
         while num_instances < num_instances_to_use:
-            group = autoscale.get_all_groups(names=[aws_group_name])[0]
-            num_instances = len(group.instances)
-            print(GetTime(),'Number of instances online:',len(group.instances))
+            instances = autoscale.describe_auto_scaling_instances()['AutoScalingInstances']
+            num_instances = len(instances)
+            print(GetTime(),'Number of instances online:', num_instances)
             sleep(3)
 
     #grab instance IDs
-    instance_ids = [i.instance_id for i in group.instances]
+    instance_ids = [i['InstanceId'] for i in instances]
     print(GetTime(),"These instances are online:",instance_ids)
 
-    instances = ec2.get_only_instances(instance_ids)
-    for instance in instances:
-        print(GetTime(),'Waiting for instance',instance.id,'to boot...')
-        while 1:
-            instance.update()
-            if instance.state == 'running':
-                print(GetTime(),instance.id,'is running!')
+    for instance_id in instance_ids:
+        print(GetTime(),'Waiting for instance',instance_id,'to boot...')
+        while True:
+            if state_name_of(instance_id, ec2) == 'running':
+                print(GetTime(),instance_id, 'is running!')
                 break
             sleep(3)
     for instance_id in instance_ids:
         print(GetTime(),'Waiting for instance',instance_id,'to report OK...')
-        while 1:
-            statuses = ec2.get_all_instance_status([instance_id])
-            if len(statuses) < 1:
-                sleep(3)
-                continue
```

Comment:
I couldn't find anything equivalent to this `get_all_instance_status` thing and I wasn't sure why you were checking it's length. I figured since I was inside the loop, if it matched `ok`, then it's okay.

Everything else should be fairly close to how it was before.